### PR TITLE
Moved vendor specific HCI to HAL

### PR DESCRIPTION
**Pull request by mtpr-ot**
Moved vendor HCI code to new HAL folder to allow out-of-tree
vendor specific HCI implementations.

### CODE REVIEW

File: include/bluetooth/hci_vs.h [hunk 1]
```diff
@@ -25,7 +26,8 @@ extern "C" {
 #define BT_HCI_VS_FW_VAR_VS_CTLR                0x0002
 #define BT_HCI_VS_FW_VAR_FW_LOADER              0x0003
 #define BT_HCI_VS_FW_VAR_RESCUE_IMG             0x0004
-#define BT_HCI_OP_VS_READ_VERSION_INFO		BT_OP(BT_OGF_VS, 0x0001)
+
+#define BT_HCI_OP_VS_READ_VERSION_INFO			BT_OP(BT_OGF_VS, 0x0001)
```
Comment (cvinayak) [inline include/bluetooth/hci_vs.h]: Please remove tabs (use spaces to align), and align with other BT_OP

File: subsys/bluetooth/controller/CMakeLists.txt [hunk 2]
```diff
@@ -60,6 +67,14 @@ zephyr_library_include_directories(
   include
   )
 
+if(CONFIG_SOC_COMPATIBLE_NRF)
+if(CONFIG_BT_HCI_VS)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/CMakeLists.txt]: indent the second if clause block

**Review by cvinayak — CHANGES_REQUESTED** [review]
Minor comments, otherwise LGTM.

**Comment by aescolar** [comment]
@mtpr-ot : Should this be closed, being replaced by #14143 ?

**Comment by mtpr-ot** [comment]
> @mtpr-ot : Should this be closed, being replaced by #14143 ?
Yes, you're right. Will close now. Thanks.

**Pull request closed without merging**